Marlin/src/gcode/gcode.cpp

```diff
@@ -823,6 +823,10 @@ void GcodeSuite::process_next_command() {
    * G-code "macros" to be called from within other G-code handlers.
    */
 
+  #if !defined(__AVR__) && !defined(strchr_P) // seems required on STM32 platform
+    #define strchr_P(s,c) strchr(s,c)
+  #endif
+
```

Comment:
We now have "`Marduino.h`" as a repository for things which are expected to be in `<Arduino.h>` and accompanying system headers, so that would be the most appropriate place to define this macro substitution.

Reply:
ah indeed, didnt think about that... will do in another PR i guess